setup.py

```diff
@@ -55,6 +56,7 @@ def build_extension(self, ext: CMakeExtension):
 
         build_args = ["--config", "Debug"] if debug else ["--config", "RelWithDebInfo"]
         configure_args = [
+            f"-DPython_EXECUTABLE={sys.executable}",
```

Comment:
```suggestion
            f"-DPython_EXECUTABLE={sys.executable}",
```

We also need `            f"-DPYTHON_EXECUTABLE={sys.executable}",`
somewhere too. Linux uses one, Windows the other

Reply:
We may need it to be:
```suggestion
            f"-DPython_EXECUTABLE={sys.executable}",  # (Windows)
            f"-DPYTHON_EXECUTABLE={sys.executable}",  # (Ubuntu)
```
We had problems with Windows in the past.

Reply:
@mlxd beat me to it.

Reply:
https://github.com/PennyLaneAI/pennylane-lightning/pull/400 for when this appeared in the past

Reply:
Ok, seems like you are correct :+1:

Reply:
Ah, its [MacOS](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/5337007792/jobs/9672499510#step:8:668) not recognizing `PYTHON_EXECUTABLE`.